### ISSUE TITLE
FF 51.0.1 - Update doesn't work when bookmark is tagged

### ISSUE THREAD

**Issue by HashtagMC** [dgutov/bmreplace#39]
In Firefox 51.0.1., updating a bookmark does not work when said bookmark has tags on it. The addon claims to have updated the name & description, when in fact it hasn't, and any further tries to update are blocked: `page already bookmarked!` This problem only occurs with tagged bookmarks, if I have the very same page bookmarked without tags, the addon does its job without complaints.

Problem occurs on Firefox 51.0.1. on Windows 7 64-bit.

**Comment by dgutov** [comment]
Hi!

Thanks for the report. What do you mean exactly when you say "claims to have updated the name & description, when in fact it hasn't"? Do you get a bookmark with the new URL but with old name and description?

It seems to work okay here in Firefox 53 (Aurora), FWIW. I created a new bookmark with a couple of tags, switches to an URL on the same domain, and pressed the shortcut. The new page because bookmarked, the name of the bookmark was correct, and the tags were there.

If you see this problem consistently, is there anything of note in the Error Console (press Ctrl+Shift+J to open it) when that happens? Please clear the existing errors before trying the scenario, by clicking on the garbage can symbol in the top left corner.

**Comment by HashtagMC** [comment]
I have bookmarks on pages whose titles and description change regularly, but URL does not. So I use the plugin to update title and description of the bookmark.

Like, `this-is-a-domain.example/page1.html` has the title "Page 1 Version 1", and a few days later the title "Page 1 Version 2". I want to update the bookmark, that has a tag, and the alerts "Do you want to update the title" (click yes) and "Do you want to update the description" (click yes) appear.

The bookmark then has _not_ been changed, but calling bookmark replace again results in an error: "This page is already bookmarked".

The browser console shows no errors.
![bookmarked already](https://cloud.githubusercontent.com/assets/13611718/23212181/d46dbb34-f905-11e6-9a27-26dd9916da44.png)

**Comment by dgutov** [comment]
Thanks.

Did it work in previous versions of Firefox, or is it the first time you've installed it?

Can you point to a public web page with description to test it on?

Does the Error Console show anything interesting?

**Comment by HashtagMC** [comment]
First time install.

I use a Greasemonkey script to add a `<meta name="description" content="[description here]">` tag to the DOM after the page is loaded. This might be the source of the issue, so I'll try to reproduce the situation with the script disabled. Hold on a sec.

**Comment by HashtagMC** [comment]
Okay, so disabling Greasemonkey doesn't work either, so that's apparently not the problem either.

The pages I use the addons on are [fanfiction.net](fanfiction.net) and [Archive of our Own](archiveofourown.org). But, as I mentioned above, their descriptions don't change by themselves.

**Comment by dgutov** [comment]
> But, as I mentioned above, their descriptions don't change by themselves.

I think I managed to reproduce the bug at first by editing the descriptions from the bookmark properties (from the bookmarks panel) and then pressing the key combination, and couple of times.

But then I ran the addon in the debugger, and I can't reproduce it anymore. :( Will have to get back to this one later. But it looks like a bug in Firefox.

**Comment by dgutov** [comment]
@HashtagMC Are you still having this problem?

**Comment by HashtagMC** [comment]
I tried a couple days ago (I've a batch of 400-some bookmarks to update) and then, updating didn't do a thing. The addon claimed to have updated the bookmark title and description, but neither(!) had changed.

**Comment by dgutov** [comment]
Could you try the latest version? I've published 1.6.3 today.

And if it still doesn't work, describe how to reproduce the problem like in https://github.com/dgutov/bmreplace/issues/40#issuecomment-315368711

**Comment by HashtagMC** [comment]
Is the updated version online for Firefox yet? As far as I know, Firefox (except for dev builds) will only accept addons signed by Mozilla, which is another step towards the end of "conventional" addons announced for end of 2017. Right now, FF says 1.6.2 - last updated January 2017.

**Comment by dgutov** [comment]
Yes: https://addons.mozilla.org/en-US/firefox/addon/bmreplace/versions/?page=1#version-1.6.3

The Extensions page has the "Check for updates" item behind the gear button, but you can install the new version manually. Also try restarting after installing it.

**Comment by HashtagMC** [comment]
Updated & restarted. Doesn't do a thing, neither name nor description are updated. Visiting a page and trying to update, the addon offers to update the name (and the description, if they don't match), then claims to have updated and thus refuses to update the bookmark again, without actually having changed something.

About reproducing the problem: I use it on stories over at fanfiction.net. I've written a script that automatically adds the description as a meta tag to the head of the page. I've disabled userscripts completely and updating still doesn't work.

Firefox is offering an update right now. I'll update FF and get back to you on that afterwards.

**Comment by dgutov** [comment]
> Visiting a page

How about telling me an actual page?

**Comment by HashtagMC** [comment]
> How about telling me an actual page?

Sure thing. You can try either of the following pages, bookmark it, manually change the description (editing the <meta name="descrption" content="description here..."> using Firefox developer tools) and get the addon to update the description to reproduce the error.

https://archiveofourown.org/works/265547
https://www.fanfiction.net/s/11382131/1

**Comment by dgutov** [comment]
The second one has no description. This addon doesn't try to "update" in in this case.

Looking at the first one now.

**Comment by HashtagMC** [comment]
> The second one has no description. This addon doesn't try to "update" in in this case.

Like I said, I add them manually (well, by script) and try to update the bookmark then (going from "no description" to "with description").

**Comment by dgutov** [comment]
> Like I said, I add them manually (well, by script)

That obviously doesn't help me to reproduce the problem, right? It *should* for for script-generated descriptions, but I'd rather not try debugging those until we determine that being script-generated is the actual problem.

**Comment by HashtagMC** [comment]
> but I'd rather not try debugging those until we determine that being script-generated is the actual problem.

Understandably. I've also tried something else: adding a description manually in the bookmark library, and then trying to update the bookmark with userscripts disabled to get the original description (first bookmark) back. Didn't work either. That better for debugging?

Also, does the addon generate any logfiles or the likes I could post here for debugging purposes? Developer console output, anything?

**Comment by dgutov** [comment]
> I've also tried something else: adding a description manually in the bookmark library, and then trying to update the bookmark with userscripts disabled to get the original description (first bookmark) back.

I have just tried that, and it worked. With a tag (randomly named) and without it. Replace "adding a description" with "changing a description", though, since there already was one there.

**Comment by HashtagMC** [comment]
... tried again now, worked *puzzled expression* Weird. But glad it works now.

**Comment by HashtagMC** [comment]
I guess it might have been the Firefox update fixing it.

**Comment by dgutov** [comment]
> Also, does the addon generate any logfiles or the likes I could post here for debugging purposes?

Nope. But generally, most problems result in exceptions being printed to the Browser Console. You can open it with Ctrl-Shift-J.

Have you tried the problem scenario with an empty Firefox profile?

**Comment by dgutov** [comment]
> I guess it might have been the Firefox update fixing it.

Cool! If it happens again, though, please try an empty profile.

**Comment by HashtagMC** [comment]
> Nope. But generally, most problems result in exceptions being printed to the Browser Console. You can open it with Ctrl-Shift-J.

> Cool! If it happens again, though, please try an empty profile.

Sure, will try both if the problem occurs again. For now, thanks for taking the time to help me :)
